test/strtoultest.c

```diff
@@ -41,11 +41,16 @@ static struct strtoul_test_entry strtoul_tests[] = {
     {
         "18446744073709551615", 0, ULONG_MAX, 1, 20
     },
+#elif !defined(__WORDSIZE) && defined(__x86_64__)
```

Comment:
I haven't tried a system with MUSL libc on AArch64, but if it doesn't define `__WORDSIZE` on x86_64 it probably doesn't do it on AArch64, so we should probably add `__aarch64__`. I don't know what other systems it supports -  `__ppc64__`?

Reply:
We coud probably do it the other way too, with just

```
#if ULONG_MAX == 4294967295
:
#else
:
#endif
```

Reply:
can we do that?  musl defines UNLONG_MAX like this
```
#define ULONG_MAX (2UL*LONG_MAX+1)
```
I guess having that work depends on the compiler preprocessor doing arithmetic evaluation

Reply:
huh, I just tested, and it seems the preprocessor is capable of arithmetic eval.  I wonder if that holds for djgpp and others, I'll push a change and we will see

Reply:
seems like it works on all our supported compilers.  thanks!